src/raftstore/store/worker/split_check.rs

```diff
@@ -125,48 +121,94 @@ impl<'a> MergedIterator<'a> {
     }
 }
 
-/// Split checking task.
-pub struct Task {
-    region: Region,
-}
+#[allow(collapsible_if)]
+fn bound_keys(
```

Comment:
I don't think all the checkers need this. 
You should let the checker handle the bounded keys itself internally. 

Btw, in our case, we don't need to check all CFs, maybe lock is enough. We can guarantee that the range for all CFs are the same.